starter-code/TicTacToe/app/src/main/java/ly/generalassemb/drewmahrt/tictactoe/GameActivity.java

```diff
@@ -1,13 +1,386 @@
+        lastPlayedLetter = last;
+    }
+
+    public void setLastPlayedPos (String last) {
+        lastPlayedPos = last;
+    }
+
+    public void setTurnText () {
+        String player;
+
+        if (turn % 2 == 0) {
+            player = player2;
+        }
+
+        else {
+            player = player1;
+        }
+
+        playerTurn.setText(player + "'s turn!");
+    }
+
+    public boolean checkA1 () {
+        if (lastPlayedLetter.equals(a2.getText().toString())) {
+            if (lastPlayedLetter.equals(a3.getText().toString())){
+                return true;
+            }
+        }
+
+        else if (lastPlayedLetter.equals(b1.getText().toString())) {
+            if (lastPlayedLetter.equals(c1.getText().toString())){
+                return true;
+            }
+        }
+
+        else if (lastPlayedLetter.equals(b2.getText().toString())) {
+            if (lastPlayedLetter.equals(c3.getText().toString())) {
+                return true;
+            }
+        }
+
+        return false;
     }
+
+    public boolean checkA2 () {
+        if (lastPlayedLetter.equals(a1.getText().toString())) {
+            if (lastPlayedLetter.equals(a3.getText().toString())) {
+                return true;
+            }
+        } else if (lastPlayedLetter.equals(b2.getText().toString())) {
+            if (lastPlayedLetter.equals(c2.getText().toString())) {
+                return true;
+            }
+        }
+
+
+        return false;
+    }
+
+    public boolean checkA3 () {
+        if (lastPlayedLetter.equals(a2.getText().toString())) {
+            if (lastPlayedLetter.equals(a1.getText().toString())) {
+                return true;
+            }
+        } else if (lastPlayedLetter.equals(b3.getText().toString())) {
+            if (lastPlayedLetter.equals(c3.getText().toString())) {
+                return true;
+            }
+        } else if (lastPlayedLetter.equals(b2.getText().toString())) {
+            if (lastPlayedLetter.equals(c1.getText().toString())) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public boolean checkB1 () {
+        if (lastPlayedLetter.equals(a1.getText().toString())) {
+            if (lastPlayedLetter.equals(c1.getText().toString())) {
+                return true;
+            }
+        } else if (lastPlayedLetter.equals(b3.getText().toString())) {
+            if (lastPlayedLetter.equals(b2.getText().toString())) {
+                return true;
+            }
+        }
+
+
+        return false;
+    }
+
+    public boolean checkB2 () {
+        if (lastPlayedLetter.equals(a1.getText().toString())) {
+            if (lastPlayedLetter.equals(b3.getText().toString())) {
+                return true;
+            }
+        } else if (lastPlayedLetter.equals(a3.getText().toString())) {
+            if (lastPlayedLetter.equals(c1.getText().toString())) {
+                return true;
+            }
```

Comment:
I think these "check" methods could be consolidated as well, then pass in references to the specific TextViews you need to call `getText()` on.